internal/lib/sandbox/namespaces.go

```diff
@@ -276,11 +285,103 @@ func (s *Sandbox) UserNsJoin(nspath string) error {
 }
 
 // PidNs specific functions
+// CreateManagedPidNamespace creates a managed pid namespace
+// it is separate from the other namespaces because pid namespaces need special handling
+// We cannot tell the runtime: "here is your pid namespace!", because it gets created
+// when the container is created and it would would be a bother having conmon (the parent of the container process)
+// unshare and then do the bind mount
+// instead, we mount the sandbox's pid namespace after the infra container is created
+// and from then on refer to it for each container create
+// thus, this should be called after the infra container has been created in the runtime
+func (s *Sandbox) CreateManagedPidNamespace(cfg *config.Config) error {
+	return s.CreateManagedPidNamespaceWithFunc(cfg, pinPidNamespace)
+}
+
+func (s *Sandbox) CreateManagedPidNamespaceWithFunc(cfg *config.Config, pinFunc func(*config.Config, string) (NamespaceIface, error)) (retErr error) {
+	if cfg == nil {
+		return errors.New("given config is nil")
```

Comment:
nit: return more context in the error message here.

Reply:
better?